docs/dev-client-networking.md

```diff
@@ -3,259 +3,647 @@ id: dev-client-networking
 
-A `NetworkSchema` indicates how to pack networked component data into `UpdateMessage` s, and has the following properties:
+### Networked Component
 
--   A `componentName` that uniquely identifies the component.
--   A `serialize` (and `deserialize`) function that defines how component data is packed into (and unpacked from) `UpdateMessage` s.
--   An optional `serializeForStorage` (and `deserializeForStorage`) function that defines how component data is packed into (and unpacked from) `StorableUpdateMessages` s, able to be saved (and loaded) from the database.
+Networked entities are any entities with the `Networked` component. The
+`Networked` component defined in
+[`src/bit-components.js`](https://github.com/mozilla/hubs/blob/master/src/bit-components.js)
+contains:
+
+- A (networked) `id`, which clients use to uniquely identify this entity across
+the network. This cannot simply be the `eid` of an entity, because `eid`s are
+assigned locally to each client.
+- A `creator`, which is used at various times to determine whether or not an
+entity should be created or removed.
+- An `owner`, which is used to determine which `client` has authority to update
+the state of this entity.
+- A `lastOwnerTime`, which is used to determine the most recent time that an
+`owner` was assigned.
+- A `timestamp`, which is the most recent time the networked state of this
+entity has been updated.
+
+The `creator` can be set to a `ClientId`, `"reticulum"`, or a `NetworkID`.
+
+- When the `creator` is a `ClientID`, it means that a client in the room has
+caused this &ldquo;root&rdquo; entity (and its descendants) to be created, and
+the entity (and its descendants) should be removed when the client leaves the
+room.
+- When the `creator` is `"reticulum"`, it means that the Reticulum is
+responsible for deciding whether this &ldquo;root&rdquo; entity should be
+created or removed.
+- When the `creator` is a `NetworkID`, it means that the entity is a descendant
+of a &ldquo;root&rdquo; entity, and its creation/removal will be subject to its
+ancestor.
+
+Conceptually, the `creator` and the `owner` act as authorities over two facets
+of a networked entity.
+
+- The `creator` is the authority over the entity&rsquo;s existence. Thus, it is
+checked when processing `CreateMessage` s and before an entity may be removed.
+For an entity to be created, its `creator` must have the appropriate
+permission. An entity&rsquo;s `creator` changes infrequently. It only happens
+when a client `pins` (or `unpins`) an entity, which changes the `creator` to
+(or from) `"reticulum"`.
+- The `owner` is the authority over the entity&rsquo;s current state. Thus, it
+is associated with `UpdateMessage`s. The `owner` is expected to change
+regularly, whenever a new client performs an action on an entity. The
+`takeOwnership` and `takeSoftOwnership` functions allow a client to establish
+itself as the `owner` of an entity.

```

Comment:
These are the only properties that are discussed in further detail and they are concepts that have partial definitions in multiple locations. Maybe these definitions can be moved under `##Ownership` and you can link to those definitions inside the short definitions in the list of properties above?

Reply:
Thanks for the comment. As written in the PR comment, this section is not ready. I will add a clearer TODO that indicates this section is WIP, or hide this section for now, and I hope we can make follow up PRs later.

Reply:
Added TODO in #160 for now, thanks